pkg/cgroups/memory.go

```diff
@@ -45,16 +44,24 @@ func (c *memHandler) Stat(ctr *CgroupControl, m *Metrics) error {
 
 	if ctr.cgroup2 {
 		memoryRoot = filepath.Join(cgroupRoot, ctr.path)
-		filenames["usage"] = "memory.current"
+		filenames["usage"] = "memory.stat"
 		filenames["limit"] = "memory.max"
 	} else {
 		memoryRoot = ctr.getCgroupv1Path(Memory)
 		filenames["usage"] = "memory.usage_in_bytes"
 		filenames["limit"] = "memory.limit_in_bytes"
 	}
-	usage.Usage, err = readFileAsUint64(filepath.Join(memoryRoot, filenames["usage"]))
-	if err != nil {
-		return err
+
+	if filenames["usage"] == "memory.stat" {
+		usage.Usage, err = readFileByKeyAsUint64(filepath.Join(memoryRoot, filenames["usage"]), "anon")
+		if err != nil {
+			return err
+		}
+	} else {
+		usage.Usage, err = readFileAsUint64(filepath.Join(memoryRoot, filenames["usage"]))
+		if err != nil {
+			return err
+		}
```

Comment:
could we move these new actions in the previous if/else statement?

We don't even need to store `filenames["usage"]` but use directly the correct filename to the new functions.  `filenames["limit"]` can be a string now, e.g. `limitFilename` instead of using a map

Reply:
Refactor done and force pushed.